cloud/terraform/terraform_controller.py

```diff
@@ -0,0 +1,112 @@
+from ast import alias
+import os
+import json
+
+
+class TerraformController:
+    def __init__(self, cloud):
+        self.cloud = cloud
+
+        if cloud == 'aws':
+            self.cloud_instance = 'aws_instance'
+        self.resources = {'resource': {self.cloud_instance: {}}}
+        self.providers = {'provider': {cloud: []}}
+
+    def __del__(self):
```

Comment:
https://stackoverflow.com/questions/1481488/what-is-the-del-method-and-how-do-i-call-it

If you want to have some kind of smart object lifecycle management, use context managers.

Reply:
I didn't know it shouldn't be used, thanks! We are going to make sure every resource gets destroyed on the main program